### PR TITLE
Remove inout-like parameters to GFS_radiation_driver

### PR DESCRIPTION
**Pull request by YakelynRJ**
This PR defined a method called _GFS_radiation_driver() that does not take zero array inputs. 

Coverage reports (updated automatically):
- test_unit: [83%](https:\/\/output.circle-artifacts.com\/output\/job\/3fc45540-4060-4207-a6ea-85734af1895f\/artifacts\/0\/tmp\/coverage\/htmlcov-test_unit\/index.html)

### CODE REVIEW

File: external/radiation/python/radiation_driver.py [hunk 1]
```diff
@@ -423,6 +421,18 @@ def _GFS_radiation_driver(
 
         scmpsw = dict()
         Diag = dict()
+        Radtend = dict()
```
Comment (nbren12) [inline external/radiation/python/radiation_driver.py:408]: No need to change this, but usually to initialize a dict one can do this: `Radtend = {}`

File: external/radiation/python/radiation_driver.py [hunk 2]
```diff
@@ -403,21 +403,11 @@ def _GFS_radiation_driver(
         cldtausw = np.zeros((IM, Model["levr"] + self.LTP))
 
         Coupling = dict()
```
Comment (nbren12) [inline external/radiation/python/radiation_driver.py:405]: Perhaps we should add some regression tests for the outputs of `Coupling` soon. It is a gap in our test coverage. Maybe you can work with @elynnwu on this.

**Review by nbren12 — APPROVED** [review]
Looks perfect. So is this function completely pure now or does it still modify some of its inputs in place?

**Comment by YakelynRJ** [comment]
@nbren12 yes, it is a pure function.

**Pull request merged**